client/src/GraphQLFileRetriever.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Libero\SharedServicesExperiment\Client;
+
+use RuntimeException;
+use GuzzleHttp\Exception\BadResponseException;
+use Softonic\GraphQL\Client as GraphQLClient;
+use Softonic\GraphQL\ResponseBuilder;
+use GuzzleHttp\ClientInterface;
+use function GuzzleHttp\Psr7\stream_for;
+
+class GraphQLFileRetriever
+{
+    /**
+     * GraphQL client instance
+     *
+     * @var GraphQLClient
+     */
+    private $client;
+    private $restFileRetriever;
+
+    public function __construct(ClientInterface $client)
+    {
+        $this->client = new GraphQLClient($client, new ResponseBuilder());
+        $this->restFileRetriever = new RestFileRetriever($client);
+    }
+
+    public function retrieveFile(string $path): FileRecord
+    {
+        $fileRecord = $this->retrieveFileMeta($path);
+
+        return $this->restFileRetriever->retrieveFile($fileRecord->getLink());
+    }
+
+    public function retrieveFileMeta(string $path): FileRecord
+    {
+        $query = <<<'QUERY'
+        query GetFileMeta(key: String) {
```

Comment:
is this something that has to be in sync with the server or can it be a subset of fields?

Reply:
you mean the key?

Reply:
or do you mean the graphql schema?

Reply:
or do you mean the schema?